### ISSUE TITLE
Use setgid in Upstart template / setuid only sets primary user group

### ISSUE THREAD

**Issue by Iristyle** [strongloop/node-foreman#10]
### TL; DR

The template for the Upstart job needs to have `setgid` (at least optionally) in addition to `setuid`

https://github.com/NodeFly/node-foreman/blob/master/lib/upstart/foreman-APP-N.conf#L11
### The longer version ;0

I'm using Capistrano to deploy.  Capistrano has kind of an architectural issue, where it expects the deploying user to be the same user as the one an application runs as. 

This is a bad practice from a security standpoint, so I have it changed up a little bit to follow a more locked-down / secure setup.  It's pretty similar to what's written in [this SO answer](http://stackoverflow.com/a/10289846/87793).
- Deploying user is the standard `ubuntu` user on our EC2 vm
- An app has it's own `foo` account, where it's only allowed read access to the directories it needs.  The exception is that the log directory can be written to.
- The `ubuntu` user and the `foo` user are both made part of the `deployers` group, so that we can actually deploy the app.
- Because `sudo` needs to be used at times during deploy, it's best done with this deploying `ubuntu` user rather than the apps `foo` user.

In any event, this setup works fine with the stock Ruby foreman, because the generated upstart script runs `exec su - foo -c "commands"`.  Node-foreman is a little more modern with its setup and uses the `setuid` stanza (available on Upstart 1.4+).

The problem with `setuid` is that it [only sets up the primary user group](https://bugs.launchpad.net/upstart/+bug/812870).  This is filed as a bug in Upstart, but the patch hasn't been integrated.

As soon as I added the `setgid` to my Upstart conf I was in business.

`setgid deployers`
### Another alternative

I don't know a ton about Upstart, but perhaps the old foreman style is generally more compatible than how yours is rigged up.  Your version is much more readable for what it's worth.
